src/components/MessageList/utils/insertDates.js

```diff
@@ -0,0 +1,39 @@
+export const insertDates = (messages) => {
+  const newMessages = [];
+  if (messages.length === 0) {
+    return newMessages;
+  }
+
+  for (const [i, message] of messages.entries()) {
```

Comment:
might be helpful to put a comment on this for loop describing what's happening for future reference